src/fielddef.ts

```diff
@@ -285,22 +286,9 @@ export function normalize(channelDef: ChannelDef<string>, channel: Channel) {
     }
 
     // Normalize bin
-    if (fieldDef.bin) {
-      const bin = fieldDef.bin;
-      if (isBoolean(bin)) {
-        fieldDef = {
-          ...fieldDef,
-          bin: {maxbins: autoMaxBins(channel)}
-        };
-      } else if (!bin.maxbins && !bin.step) {
-        fieldDef = {
-          ...fieldDef,
-          bin: {
-            ...bin,
-            maxbins: autoMaxBins(channel)
-          }
-        };
-      }
+    const bin = normalizeBin(fieldDef.bin, channel);
```

Comment:
1) Why should you call this no matter fieldDef.bin is truthy or not?  The predicate for if should be `fieldDef.bin` and `normalizeBin` should be called inside.  
2) Setting `fieldDef.bin` cause side effect.  Why don't you follow the pattern we had earlier?